spec/solver_spec.rb

```diff
@@ -0,0 +1,36 @@
+require './solver'
+
+solver = Solver.new
+describe 'factorial tests' do
+  it 'should raise error for negative numbers' do
+    expect { solver.factorial(-1) }.to raise_error(ArgumentError)
+  end
+  it 'should return 1 if num = 0' do
+    expect(solver.factorial(0)).to eql(1)
+  end
+  it 'should return the correct factorial' do
+    expect(solver.factorial(5)).to eq(120)
+    expect(solver.factorial(6)).to eq(720)
+    expect(solver.factorial(7)).to eql(5040)
+  end
+end
+
+describe 'Reverse tests' do
+  it 'should reverse a string' do
+    expect(solver.reverse_string('Hello')).to eql('olleH')
+  end
+  it 'should reverse a string' do
+    expect(solver.reverse_string('ahmed')).to eql('demha')
+  end
+end
+describe 'fizzbuzz Tests' do
+  it 'should return fizzbuzz' do
+    expect(solver.fizzbuzz(15)).to eql('fizzbuzz')
+  end
+  it 'should return fizz' do
+    expect(solver.fizzbuzz(9)).to eql('fizz')
+  end
+  it 'should return buzz' do
+    expect(solver.fizzbuzz(25)).to eql('buzz')
+  end
```

Comment:
Great job team 🤝

- [ ] As stated in the project requirements, kindly ensure the include a test case for the `fizzbuzz` method where `N is not divisible by 3 or 5`.
![fizzbuzz](https://github.com/ahmed025566/TDD/assets/63917725/97de3ad0-515b-4e64-8d7b-9535352f4a03)